src/common/utils/auth_tests.rs

```diff
@@ -31,94 +51,2254 @@ mod tests {
+                bypass_check: false,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    // Tests for routes defined in handler::http::request::alerts::deprecated.
+
+    #[tokio::test]
+    async fn save_alert() {
+        test_auth(
+            Method::POST,
+            format!("api/{ORG_ID}/{STREAM_NAME}/alerts"),
+            AuthExtractor {
+                auth: AUTH_HEADER_VAL.to_string(),
+                method: format!("{POST_METHOD}"),
+                o2_type: format!("alert:{ORG_ID}"),
+                org_id: format!("{ORG_ID}"),
+                bypass_check: false,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn update_alert() {
+        test_auth(
+            Method::PUT,
+            format!("api/{ORG_ID}/{STREAM_NAME}/alerts/{ALERT_NAME}"),
+            AuthExtractor {
+                auth: AUTH_HEADER_VAL.to_string(),
+                method: format!("{PUT_METHOD}"),
+                o2_type: format!("alert:{ALERT_NAME}"),
+                org_id: format!("{ORG_ID}"),
+                bypass_check: false,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn list_stream_alerts() {
+        test_auth(
+            Method::GET,
+            format!("api/{ORG_ID}/{STREAM_NAME}/alerts"),
+            AuthExtractor {
+                auth: AUTH_HEADER_VAL.to_string(),
+                method: format!("{LIST_METHOD}"),
+                o2_type: format!("alert:{ORG_ID}"),
+                org_id: format!("{ORG_ID}"),
+                bypass_check: false,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn list_alerts() {
+        test_auth(
+            Method::GET,
+            format!("api/{ORG_ID}/alerts"),
+            AuthExtractor {
+                auth: AUTH_HEADER_VAL.to_string(),
+                method: format!("{LIST_METHOD}"),
+                o2_type: format!("alert:{ORG_ID}"),
+                org_id: format!("{ORG_ID}"),
+                bypass_check: false,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn get_alert() {
+        test_auth(
+            Method::GET,
+            format!("api/{ORG_ID}/{STREAM_NAME}/alerts/{ALERT_NAME}"),
+            AuthExtractor {
+                auth: AUTH_HEADER_VAL.to_string(),
+                method: format!("{GET_METHOD}"),
+                // Is this correct? Can two streams have the same name if they
+                // have different types?
+                o2_type: format!("{STREAM_NAME}:alerts"),
+                org_id: format!("{ORG_ID}"),
+                bypass_check: false,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
```

Comment:
?

Reply:
Does not seem to be correct, need to check